### ISSUE TITLE
Suggestion: Paste and Go

### ISSUE THREAD

**Issue by Chewbugga** [marklieberman/foxygestures#143]
It's basically a function I've known from Opera since forever and been able to replicate via FireGestures.
It simply pastes the current clipboard content into the address bar and executes.
In terms of keyboard shortcuts Ctrl+L / Ctrl+V / RETURN

Any solutions and/or suggestions greatly appreciated!
Great Addon, huge potential!

**Comment by ganego** [comment]
Came here for the same reason. Back in FireGestures you could use "readFromClipboard();" in a custom script but I guess this function was provided by FireGestures. And I guess Mozilla does not care enough to allow this again, correct?

**Comment by marklieberman** [comment]
The Web Extensions APIs for [clipboards](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Interact_with_the_clipboard#Reading_from_the_clipboard) are kinda incomplete. Its doable but in a really lousy way that requires injecting a textbox into the DOM to read the clipboard. Basically the reverse of [Copy URL to Clipboard](https://github.com/marklieberman/foxygestures/wiki/List-of-Common-User-Scripts#copy-url-to-clipboard). You cannot do it yet though because I'd have to request clipboardRead permission in manifest.json.

**Comment by ganego** [comment]
Yes what I have right now is the following, and I also set the clipboardRead permisson in the manifest and loaded the addon into my nightly with disabled signature checking.

```
// Create a text area with the value we want to copy.
let textarea = document.createElement("textarea");

// Style textarea to make it fairly invisible.
// Copy doesn't work when display: none.
textarea.style.maxHeight = '2em';
textarea.style.maxWidth = '2em';
textarea.style.position = 'fixed';
textarea.style.top = 0;
textarea.style.left = 0;
textarea.style.padding = 0;
textarea.style.border = 'none';
textarea.style.outline = 'none';
textarea.style.boxShadow = 'none';
textarea.style.background = 'transparent';

textarea.contentEditable="true";

// Append the textarea, copy its value, then remove it.
document.body.appendChild(textarea);
textarea.select();
//textarea.focus()
document.execCommand("Paste");
console.log(textarea.value);
console.log(textarea.textContent);
//textarea.remove();
```

Little problem: It does not work.

**Comment by marklieberman** [comment]
Even weirder... here is a version of the script that works but only once. If I use "Load temporary addon" I can paste whatever was in my clipboard before the addon was loaded. But it only works once, then never again until the addon is reloaded.

```js
executeInBackground(() => {
    // Create a text area with the value we want to copy.
    let textarea = document.createElement("textarea");
    textarea.contentEditable = true;
    document.body.appendChild(textarea);
    textarea.select();
    document.execCommand("paste");
    console.log(textarea.value);
    textarea.parentNode.removeChild(textarea);
});
```

**Comment by ganego** [comment]
I can somewhat reproduce this, it worked twice but I had to restart and load the foxygestures options or so. This is definitely not working as it should

**Comment by marklieberman** [comment]
clipboard.readText and clipboard.writeText appear to have landed in FF63 so it may be worth revisiting this issue soon.
